File: app/src/main/java/rubik/shifttest/presentation/MainActivity.java
package rubik.shifttest.presentation;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.content.SharedPreferences;
import android.os.Bundle;

import rubik.shifttest.R;
import com.rubik.shifttest.data.data.storage.sharedprefs.SharedPrefUserStorage;

public class MainActivity extends AppCompatActivity {

    private static final String SIGN_UP_FRAGMENT_TAG = "signUpFragment";
    private static final String GREETING_FRAGMENT_TAG = "greetingFragment";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        FragmentManager fragmentManager = getSupportFragmentManager();

        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        SharedPreferences sharedPrefs = getSharedPreferences(SharedPrefUserStorage.SHARED_PREFS_USER_REGISTER_CREDENTIAL, MODE_PRIVATE);
        if(sharedPrefs.contains(SharedPrefUserStorage.KEY_USER_REGISTER_CREDENTIAL)) {
            GreetingFragment greetingFragment = new GreetingFragment();
            fragmentTransaction.add(R.id.fragment_container, greetingFragment, GREETING_FRAGMENT_TAG);
            fragmentTransaction.commit();
        } else {
            SignUpFragment signUpFragment = new SignUpFragment();
            fragmentTransaction.add(R.id.fragment_container, signUpFragment, SIGN_UP_FRAGMENT_TAG);
            fragmentTransaction.commit();
        }
    }
}

File: app/src/main/java/rubik/shifttest/ui/MainActivity.kt
package rubik.shifttest.ui

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.NavController
import androidx.navigation.findNavController
import ru.rubik.navigation.NavCommand
import ru.rubik.navigation.NavCommands
import ru.rubik.navigation.NavigationProvider
import ru.rubik.user.data.storage.sharedprefs.SharedPrefUserStorage
import rubik.shifttest.R

class MainActivity : AppCompatActivity(), NavigationProvider {

	private val navController: NavController
		get() = findNavController(R.id.containerView)

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_main)
	}

	override fun onResume() {
		super.onResume()
		val sharedPrefs = getSharedPreferences(SharedPrefUserStorage.SHARED_PREFS_USER_REGISTER_CREDENTIAL, MODE_PRIVATE)
		if (sharedPrefs.contains(SharedPrefUserStorage.KEY_USER_REGISTER_CREDENTIAL)) {
			navController.navigate(R.id.action_registrationFragment_to_greetingFragment)
		}
	}

	override fun launch(navCommand: NavCommand) {
		when (val target = navCommand.target) {
			is NavCommands.DeepLink -> navController.navigate(target.url)
			else                    -> throw IllegalArgumentException("Navigation error while launch")
		}
	}
}